py/gc.c

```diff
@@ -114,7 +114,7 @@
 
 #if MICROPY_PY_THREAD && !MICROPY_PY_THREAD_GIL
 #define GC_MUTEX_INIT() mp_thread_recursive_mutex_init(&MP_STATE_MEM(gc_mutex))
-#define GC_ENTER() mp_thread_recursive_mutex_lock(&MP_STATE_MEM(gc_mutex), 1)
+#define GC_ENTER() mp_thread_recursive_mutex_lock(&MP_STATE_MEM(gc_mutex), -1)
```

Comment:
Is it worth defining a constant for this `-1`, to make it clear what it means?  Eg `MP_THREAD_MUTEX_TIMEOUT_FOREVER`.